File: c++/Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        reverse(digits.begin(), digits.end());
        int i = 1;
        int carry = (digits[0]+1)/10;
        digits[0] = (digits[0]+1)%10;
        while(i < digits.size()){
            if(carry == 0)
                break;
            int c = (digits[i]+carry)/10;
            digits[i] = (digits[i]+carry)%10;
            carry = c;
            i++;
        }
        if(carry!= 0)
            digits.push_back(carry);
        reverse(digits.begin(), digits.end());
        return digits;
    }
};

File: Java/Plus One.java
//https://leetcode.com/problems/plus-one/

class Solution {
    public int[] plusOne(int[] digits) {
        int add = 1;
        for(int i = digits.length -1; i >=0 ; i--){
            int temp = digits[i] + add;
            if(temp !=10){
                digits[i] = temp;
                add = 0;
                break;
            }
            else{
                digits[i] = 0;
                add = 1;
            }   
        }
        if(add == 1){
            int[] newDigits= new int[digits.length + 1];
            newDigits[0] = 1;
            for(int i = 0; i < digits.length ; i++){
                newDigits[i+1] = digits[i];
            }
            return newDigits;
        }
        return digits;        
    }
}